src/main/java/kita/Commands.java

```diff
@@ -0,0 +1,273 @@
+        assert tasks != null;
+        assert saveSystem != null;
+
+        this.tasks = tasks;
+        this.saveSystem = saveSystem;
+    }
+
+    /**
+     * Appends to a given StringBuilder and also prints the output
+     * @param sb
+     * @param inputStr
+     */
+    private void printAndAppend(StringBuilder sb, String inputStr) {
+        sb.append(inputStr + "\n");
+        System.out.println(inputStr);
+    }
+
+    /**
+     * Prints a standard divider line
+     * @returns The output string for the GUI
+     */
+    public String printLine() {
+        StringBuilder output = new StringBuilder();
+        printAndAppend(output, "____________________________________________________________");
+        return output.toString();
+    }
+
+    /**
+     * Adds a task to the TaskList stored in this Commands instance
+     * Also writes to Storage
+     *
+     * @param newTask Task to add
+     * @returns The output string for the GUI
+     */
+    public String addTask(Task newTask) throws IOException {
+        this.tasks.addTask(newTask);
+        saveSystem.writeTasksToFile(this.tasks.getAllTasks());
+        StringBuilder output = new StringBuilder();
+        printAndAppend(output, "Got it. I've added this task: ");
+        printAndAppend(output, "  " + newTask);
+        printAndAppend(output, "Now you have " + this.tasks.size() + " tasks in the list.");
+        return output.toString();
+    }
+
+    /**
+     * Creates an "Event" Task
+     * Also writes to Storage
+     *
+     * @param eventMatcher The Matcher object used to match the "event" creation
+     * @returns The output string for the GUI
+     */
+    public String createEvent(Matcher eventMatcher) throws IOException {
+        String from = eventMatcher.group("from");
+        String to = eventMatcher.group("to");
+        if (from == null) {
+            from = eventMatcher.group("from2");
+            to = eventMatcher.group("to2");
+        }
+        Task newTask = new Event(eventMatcher.group("name"), from, to);
+        return this.addTask(newTask);
+    }
+
+    /**
+     * Creates a "Deadline" Task
+     * Also writes to Storage
+     *
+     * @param deadlineMatcher The Matcher object used to match the "deadline" creation
+     * @returns The output string for the GUI
+     */
+    public String createDeadline(Matcher deadlineMatcher) throws IOException {
+        Task newTask = new Deadline(deadlineMatcher.group(1), deadlineMatcher.group(2));
+        return this.addTask(newTask);
+    }
+
+    /**
+     * Creates a "Todo" Task
+     * Also writes to Storage
+     *
+     * @param todoMatcher The Matcher object used to match the "todo" creation
+     * @returns The output string for the GUI
+     */
+    public String createToDo(Matcher todoMatcher) throws IOException {
+        Task newTask = new ToDo(todoMatcher.group(1));
+        return this.addTask(newTask);
+    }
+
+    /**
+     * Marks >= 1 Tasks as "completed"

```

Comment:
I dont quite understand what the JavaDoc is trying to say, perhaps the comment can be made clearer?